### PR TITLE
Pre release/3.0.0 rc1

### PR DESCRIPTION
**Pull request by repjarms**
### Summary

This adds the updated api definition and new auth types to the SDK. This is an WIP due to some outstanding changes that need to be made:

- STT WS support with the new auth scheme
- CoreML authentication issues

I am opening this now so that the main changes can be reviewed.

### CODE REVIEW

File: Source/AssistantV1/Models/DialogNode.swift [hunk 1]
```diff
@@ -23,7 +23,7 @@ public struct DialogNode: Codable, Equatable {
     /**
      How the dialog node is processed.
      */
-    public enum NodeType: String {
+    public enum TypeEnum: String {
```
Comment (mkistler) [inline Source/AssistantV1/Models/DialogNode.swift]: Looks like we now generate multiple `TypeEnum` enums.  We should make sure that these are usable, by e.g. qualifying them with the struct name.

File: Source/AssistantV1/Models/DialogNodeNextStep.swift [hunk 2]
```diff
@@ -40,7 +40,7 @@ public struct DialogNodeNextStep: Codable, Equatable {
          - `reprompt`
          - `skip_slot`
          - `skip_all_slots`
-     If you specify `jump_to`, then you must also specify a value for the `dialog_node` property.
+          If you specify `jump_to`, then you must also specify a value for the `dialog_node` property.
```
Comment (mkistler) [inline Source/AssistantV1/Models/DialogNodeNextStep.swift]: This reformatting is strange.

File: Source/AssistantV1/Models/DialogSuggestionResponseGeneric.swift [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+    /**
+     The text of the response.
+     */
+    public var text: String?
+
+    /**
+     How long to pause, in milliseconds.
+     */
+    public var time: Int?
```
Comment (mkistler) [inline Source/AssistantV1/Models/DialogSuggestionResponseGeneric.swift]: This is a terrible name.  I know this comes from the API doc, but I wonder if it is still possible to change it.

**Comment by mkistler** [comment]
It might be better to break this PR into one that deals only with the copyright changes, since there appear to be many of these, and another with the new generation changes.

File: Source/AssistantV1/Models/DialogSuggestionResponseGeneric.swift (hunk 3 above)
Comment (repjarms) [reply Source/AssistantV1/Models/DialogSuggestionResponseGeneric.swift]: @mkistler I don't think we are going to be able to change this for now. We should discuss with the docs team.

**Comment by repjarms** [comment]
@mkistler I've pushed commits addressing the WS support changes and CoreML. The PR is now ready for full review.

File: Source/AssistantV1/Models/DialogNode.swift (hunk 1 above)
Comment (repjarms) [reply Source/AssistantV1/Models/DialogNode.swift]: I believe that all of the `TypeEnum` definitions are properties of a containing struct. I am not sure that we will have collision here, and all test cases build correctly.

**Pull request closed without merging**